File: 2025/week2/pset2/readability/readability.c
/*
    Week2-PSET2: Readability:
    start time: 2023-11-01T13:11:48IN
*/

#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int readability(string);

int main(void)
{
    string text;

    // User Input
    do
    {
        text = get_string("Text: ");
    }
    while (!(isalpha(text[0])));

    // Getting Grade / Index
    int grade = readability(text);

    // Comparing Grades
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

    return (0);
}

/* Function to calculate readability */
int readability(string text)
{
    // Initializing Variables
    int length = strlen(text);
    int letters, words, sentences;
    float L, S;

    // Setting Pre-values
    letters = sentences = L = S = 0;
    words = 1;

    // Counting Letters, Words & Senentces
    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }

        if (text[i] == ' ')
        {
            words += 1;
        }

        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences += 1;
        }
    }

    /* Coleman-Liau Index Logic*/

    // Calculation L and S.
    L = ((float) letters / (float) words) * (100.0);
    S = ((float) sentences / (float) words) * (100.0);

    // formula implimentation
    float index = (0.0588 * L) - (0.296 * S) - 15.8;

    return (round(index));
}

// dur-03:08:41

File: 2025/week6/pset6/sentimental-readability/readability.py
# Coleman-Liau Index : Index = 0.0588 * L - 0.296 * S - 15.8
# L is the average number of letters per 100 word
# S is the average number of sentences per 100 words in the text.

# count the number of letters, words, and sentences in the text
# if A-Z, a-z -> letter
# if seperated by space -> word
# if seperated by . ! ?  -> sentence

# if grade index > 16 -> Grade 16+
# if grade index < 1 -> Before Grade 1

# TODO:
# 1. get user input
# 2. count letters
# 3. count words
# 4. count sentences
# 5. cal index using formula

import cs50

# vars
letters = 0
words = 1
sentences = 0

# input
text = cs50.get_string("Text: ")

# cals letters, words, sentences
for l in text:
    if l.isalpha():
        letters += 1
    elif l == " ":
        words += 1
    elif l == "!" or l == "?" or l == ".":
        sentences += 1

# the index
L = letters / words * 100
S = sentences / words * 100

index = round(0.0588 * L - 0.296 * S - 15.8)

# printing results
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
